File: LeetCode/p455.cpp
class Solution {
public:
    int findContentChildren(vector<int> &g, vector<int> &s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int child_idx = 0;
        for (int cookie_idx = 0; cookie_idx < s.size(); ++cookie_idx) {
            if (child_idx == g.size())
                break;
            if (s[cookie_idx] >= g[child_idx])
                ++child_idx;
        }
        return child_idx;
    }
};

File: LeetCode/p455.rs
impl Solution {
  pub fn find_content_children(mut g: Vec<i32>, mut s: Vec<i32>) -> i32 {
    g.sort_unstable();
    s.sort_unstable();
    let mut iter = s.into_iter();
    g.into_iter().fold(0, |acc, x| {
      while let Some(i) = iter.next() {
        if i >= x {
          return acc + 1;
        }
      }
      acc
    })
  }
}

